sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_deployment.py

```diff
@@ -212,6 +217,14 @@ def _to_rest_object(self, location: str) -> BatchDeploymentData:  # pylint: disa
             properties=self.properties,
         )
 
+        if is_private_preview_enabled() and self.job_definition:
+            non_flat_data = {}
+            non_flat_data["component_deployment"] = self.job_definition._to_dict()
+            flat_data = flatten(non_flat_data, ".")
+            flat_data_keys = flat_data.keys()
+            for k in flat_data_keys:
```

Comment:
So we append and overwrite `self.properties` but if there are other things inside `self.properties`, we're ok to keep those things?
Or, should we just do `self.properties = flat_data` ?

Reply:
I think it must be append+overwrite but not complete overwrite based on the `is_private_preview_enabled` flag if I'm not mistaken?

Reply:
That is correct, if private preview is enabled we will append the flattened component deployment values that are in flat_data